src/components/moderation/ModerationInfoModal.tsx

```diff
@@ -0,0 +1,188 @@
+  )
+
+  const { data } = getBlockedInPostIdDetailedQuery.useQuery(chatId)
+  const blockedUsers = data?.address ?? []
+  const blockedUsersCount = blockedUsers.length
+
+  const { mutate } = useCommitModerationAction()
+
+  if (!myAddress) return null
+
+  const cardData: DataCardProps['data'] = blockedUsers.map((blockedData) => {
+    const address = blockedData.resourceId
+    const reasonText = blockedData.reason.reasonText
+    return {
+      title: address,
+      customContent: (
+        <DataCardContent
+          address={address}
+          reasonText={reasonText}
+          onUnblock={() =>
+            dispatch({
+              type: 'open',
+              payload: {
+                id: address,
+                reasonText,
+              },
+            })
+          }
+        />
+      ),
+    }
+  })
+
+  const unblock = () => {
+    if (!toBeUnblocked) return
+    mutate({
+      action: 'unblock',
+      resourceId: toBeUnblocked.id,
+      address: myAddress,
+      ctxPostId: chatId,
+    })
+  }
+
+  return (
+    <>
+      <Modal
+        {...props}
+        isOpen={props.isOpen && !isOpenConfirmation}
+        title='🛡 Moderation'
+        description='Moderated content will not be deleted from the blockchain but be hidden from the other users in Grill.chat.'
+      >
+        <div className='flex flex-col gap-2'>
+          <span className='text-sm text-text-muted'>
+            Blocked users ({blockedUsersCount})
+          </span>
+          <div className='overflow-hidden rounded-2xl bg-background-lighter px-4 pr-1'>
+            {blockedUsersCount ? (
+              <DataCard
+                className='max-h-96 overflow-y-scroll rounded-none p-0 py-4 pr-1 scrollbar-track-background-lighter scrollbar-thumb-background-lightest/70'
+                data={cardData}
+              />
+            ) : (
+              <div className='flex flex-col items-center gap-4 py-4 text-center'>
+                <Image src={BlockedImage} alt='' />
+                <span className='text-text-muted'>
+                  There&apos;re no blocked users yet.
+                </span>
+              </div>
+            )}
+          </div>
+        </div>
+      </Modal>
+      <ConfirmationModal
+        isOpen={isOpenConfirmation}
+        closeModal={() => dispatch({ type: 'close' })}
+        title='🤔 Unblock user'
+        primaryButtonProps={{ children: 'Yes, unblock', onClick: unblock }}
+        secondaryButtonProps={{ children: 'No, keep blocked' }}
+        content={() => {
+          if (!toBeUnblocked) return null
+          return (
+            <DataCard
+              data={[
+                {
+                  title: toBeUnblocked?.id,
+                  customContent: (
+                    <DataCardContent
+                      address={toBeUnblocked.id}
+                      reasonText={toBeUnblocked.reasonText}
+                    />
+                  ),
+                },
+              ]}
+            />
```

Comment:
I think it will be better to extract it to the separate component

Reply:
for this one I think its okay because its a simple enough function imo